### PR TITLE
Implement StatelessLSTM

### LINKED ISSUE #911: Stateless LSTM link
I need a stateless-LSTM link like GRU. Stateful-LSTM link makes a code simple, but it is difficult to generate a sequence with beam search because we need to store top-k states to a priority queue.

We also need to design consistency of names.

- Stateful LSTM: `links.LSTM`
- Stateless LSTM: new!
- Stateful GRU: `links.StatefulGRU`
- Stateless GRU: `links.GRU`

### PR DESCRIPTION
**Pull request by unnonouno**
fix #911

### CODE REVIEW

File: chainer/links/connection/lstm.py [hunk 1]
```diff
@@ -7,7 +7,75 @@
...
+        out_size (int): Dimensionality of output vectors.
+
+    Attributes:
+        upward (chainer.links.Linear): Linear layer of upward connections.
+        lateral (chainer.links.Linear): Linear layer of lateral connections.
+
+    """
+
+    def __call__(self, c, h, x):
+        """Updates the internal state and returns the LSTM outputs.
```
Comment (beam2d) [inline chainer/links/connection/lstm.py]: This summary might be confusing a bit, since there is no internal state in StatelessLSTM.

**Comment by beam2d** [comment]
Can you add an entry of StatelessLSTM to the Sphinx document?

**Comment by unnonouno** [comment]
I updated the document.

**Comment by beam2d** [comment]
LGTM!

**Pull request merged**